### PR TITLE
Use dataset random_shuffle() for shuffle nightly

### LINKED ISSUE #32272: [Ray 2.3 Release] `shuffle_1tb_5000_partitions` failing with OOM-related errors
Failed twice in a row on master branch on Jan 31.

I suspect it's the same issue as https://github.com/ray-project/ray/issues/32195 and https://github.com/ray-project/ray/issues/32203

* BuildKite: https://buildkite.com/ray-project/release-tests-branch/builds/1321#018604d7-86a6-4d31-be0f-2f184274d52f
* Anyscale: https://console.anyscale-staging.com/o/anyscale-internal/projects/prj_qC3ZfndQWYYjx2cz8KWGNUL4/clusters/ses_hkf1p9x61hkd22ysp87bjsh6ac?command-history-section=command_history

```
ray.exceptions.OwnerDiedError: Failed to retrieve object 00c24d5df3b26aefffffffffffffffffffffffff03000000ea240000. To see information about where this ObjectRef was created in Python, set the environment variable RAY_record_ref_creation_sites=1 during `ray start` and `ray.init()`.

The object's owner has exited. This is the Python worker that first created the ObjectRef via `.remote()` or `ray.put()`. Check cluster logs (`/tmp/ray/session_latest/logs/*932061f3dc6ded9ee31ec7d808c93f35e0d4a01ab45e04fbe1d24277*` at IP address 172.31.141.102) for more information about the Python worker failure
```

### PR DESCRIPTION
**Pull request by jianoaix**
## Why are these changes needed?
The current test is running python/ray/experimental/shuffle.py, which has the issue:
- Not reliable as objects may not be owned by driver (if those nodes crashes, the entire job may fail if objects got lost)
- It's not really testing the target code path (i.e. Dataset)

This is also to resolve #32272

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/release_tests.yaml [hunk 1]
```diff
@@ -3535,27 +3535,6 @@
...
-- name: shuffle_1tb_5000_partitions
-  group: core-multi-test
-  working_dir: nightly_tests
-  legacy:
-    test_name: shuffle_1tb_5000_partitions
-    test_suite: nightly_tests
-
-  frequency: nightly-3x
-  team: core
-  env: staging
```
Comment (c21) [inline release/release_tests.yaml:3547]: just to double check - we don't need `env: staging` now?

**Review by c21 — APPROVED** [review]
LG w/ one question.

**Comment by cadedaniel** [comment]
Can we kick off a run to make sure this passes?

**Comment by jianoaix** [comment]
> Can we kick off a run to make sure this passes?

I simply removed this test, because there is already a `dataset_shuffle_random_shuffle_1tb` exist for Dataset random shuffle. We do not need to waste resource for duplicate test.

**Comment by rkooo567** [comment]
Wait can we revert this change? This test is not to test dataset but to test the core (e.g., how it works when there are so many objects. That's why we kept 5000 partitions while we deleted all 1000 partition tests). The test also belongs to the core team. Let's not delete other team's release tests without discussing with them.

**Comment by rkooo567** [comment]
Or do you guys have workloads that actually create 5000 partitions (25M objects)?

**Comment by jianoaix** [comment]
I briefly discussed with @ericl on whether https://sourcegraph.com/github.com/ray-project/ray@master/-/blob/release/nightly_tests/shuffle/shuffle_test.py is relevant. For owner of this test, I think we have been changing the team owner to data team recently. So I didn't realize this is still relevant and just stress testing just core? If so we can revert it. One thing to note is that this test creates objects owned by non-driver node so it's more vulnerable to node failures.

There is no data test with this exact config.

**Comment by rkooo567** [comment]
Yeah to be clear, we removed every other shuffle test and only left "autoscaling" & "chaos" & and this test for testing object stores (with core). I think the fact that there was an OOM killer indicates something bad (this test is not supposed to OOM). I guess it was a miscommunication when changing the ownership of tests.

**Comment by rkooo567** [comment]
More details were https://docs.google.com/document/d/1dyxwOUNWqn8GEV0iy-onp55Tgy89jakvdxwBUM5VXAs/edit#. 
 
I will be in charge of reverting this + future investigation.

**Comment by jianoaix** [comment]
I see. Makes sense to revert. Please prioritize it, the bug (https://github.com/ray-project/ray/issues/32272) is a release blocker.

**Pull request merged**